### PR TITLE
20.7 fb elisa vis 7808

### PR DESCRIPTION
**Pull request by labkey-klum**
#### Rationale
This is a follow on story to the initial [ELISA assay improvements](https://docs.google.com/document/d/131Mh8D1HMs89YltSr263w931cbsqBi96B--xmTVu4OU/edit) which added support for multi-plate, multi-well data formats. Some work was done on the StatsService to allow more general handling of curve fit options between assay providers as well as methods to serialize/deserialize fit parameters to JSON.

This PR also introduces a new four parameter fit which is better adapted to handle the types of data we expect to see from the new ELISA changes. The algorithm uses a hybrid simplex-based approach that will evaluate fit errors between two optimization algorithms:
- Nelder-Mead
- Powell
And select the results which use the best fit.

#### Related Pull Requests
https://github.com/LabKey/commonAssays/pull/245

#### Changes
- Four parameter simplex curve fit
- JSON serialization support for curve fits
- AssayPlateProvider support for declaring which types of curve fits are available

### CODE REVIEW

File: core/src/org/labkey/core/statistics/FourParameterSimplex.java [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+                optimize(powellOptimizer, start);
+            }
+            catch (TooManyEvaluationsException e)
+            {
+                _log.info(e.getMessage());
+            }
+        }
+
+        _log.info("optimizer : " + _lastOptimizer + " fit error : " + calculateFitError(createParams(_bestFit)));
+        _log.info("params " + createParams(_bestFit).toMap());
```
Comment (cnathe) [inline core/src/org/labkey/core/statistics/FourParameterSimplex.java:80]: Do you want to leave in this log info?

File: core/src/org/labkey/core/statistics/FourParameterSimplex.java [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+                simplex,
+                GoalType.MINIMIZE,
+                new InitialGuess(start));
+
+        if (optimum.getValue() < _minError)
+        {
+            _minError = optimum.getValue();
+            _bestFit = optimum.getPoint().clone();
+            _lastOptimizer = optimizer;
+            //_log.info("optimizer: " + optimizer.toString() + " fit error : " + _minError + " evals: " + optimizer.getEvaluations());
```
Comment (cnathe) [inline core/src/org/labkey/core/statistics/FourParameterSimplex.java]: Leave in this log info or remove?

**Review by cnathe — APPROVED** [review]

File: core/src/org/labkey/core/statistics/FourParameterSimplex.java (hunk 1 above)
Comment (labkey-klum) [reply core/src/org/labkey/core/statistics/FourParameterSimplex.java:80]: yes, keep for now

File: core/src/org/labkey/core/statistics/FourParameterSimplex.java (hunk 2 above)
Comment (labkey-klum) [reply core/src/org/labkey/core/statistics/FourParameterSimplex.java]: I'll remove this one.

**Pull request merged**